src/evaluator/functions.ts

```diff
@@ -441,11 +441,44 @@ diff.changedOnly = () => {
 }
 diff.changedOnly.arity = 3
 
+const array: FunctionSet = {}
+
+array.unique = async function (args, scope, execute) {
+  if (args.length > 1) {
+    throw new Error('Incorrect number of arguments to function "array::unique"')
+  }
+
+  const value = await execute(args[0], scope)
+  if (!value.isArray()) {
+    return NULL_VALUE
+  }
+
+  const result = []
+  for await (const iter of value) {
+    const element = await iter.get()
+    switch (iter.type) {
+      case 'number':
+      case 'string':
+      case 'boolean':
+      case 'datetime':
+        if (result.includes(element) === false) {
+          result.push(element)
+        }
+        break
+      default:
+        result.push(element)
+    }
+  }
```

Comment:
Avoid linear search?

```suggestion
  const result = []
  const added = new Set()
  for await (const iter of value) {
    const element = await iter.get()
    switch (iter.type) {
      case 'number':
      case 'string':
      case 'boolean':
      case 'datetime':
        if (!added.has(element.data)) {
          added.add(element.data)
          result.push(element)
        }
        break
      default:
        result.push(element)
    }
  }
```

Reply:
Perf! Good point